### DISCUSSION TITLE
Accessing the Return-Tracks property of the currently active Track

### DISCUSSION THREAD

**Discussion by chozilla** [leolabs/ableton-js#79] [Q&A] [answered]
Hello,

I have heard that the mixer features are currently limited and hard to implement. 

I am looking for a way to detect the selected track in a live set in the users view (should be this one: `Live.Song.Song.View.selected_track`) and to manipulate the return_tracks for that one track relatively. So when I change the active track i can manipulate the next set of return_tracks (A, B, C, D... in Ableton). However I can not find how I would change the send % to a return-track of a track.

thanks for the help
best regards
 \ chozilla

**Comment by leolabs** [comment, marked as answer]
Hey @chozilla,

I've looked at the [unofficial API docs](https://nsuspray.github.io/Live_API_Doc/11.0.0.xml#Live-MixerDevice-MixerDevice) and it seems that unfortunately the `sends` property of the `mixer_device` of tracks is read-only.

If you'd like this to change, you can send a message to Ableton's support team and ask for the addition of this feature.

**Reply by leolabs** [reply]
I stand corrected, the properties are actually DeviceParameters and can be observed and changed. I've pushed a new version to support this :)

**Reply by chozilla** [reply]
Thank you so much! 
This is really the first step into a better workflow for me.

Not sure if i need to improve my 'await' game to cache something... 
But it works for now! and I got control over my DAW relative to the selected Track/Scene/Clip. 

This levels up my Midi devices and makes things possible that only worked on the Push before.

![image](https://user-images.githubusercontent.com/4330455/212522825-bf10f7e3-4528-4373-adba-c28851e95834.png)

**Reply by leolabs** [reply]
Your await game looks perfectly fine :)

Since the communication between ableton-js and the MIDI script runs locally, the latency is low enough that all these `await`s don't take too long. I'm currently thinking about ways to simplify the syntax a bit so you could chain these requests, like:

```ts
const sends = await ableton.song.view.get("selected_track").get("mixer_device").get("sends");
```

But I haven't quite figured out the best approach yet. Optimally, those chain requests would be collected by ableton-js and sent to the MIDI script as one request...

**Reply by chozilla** [reply]
I mean you would need to extend the Promise to have a scoped .get() function that is context sensitive if you want to stay compatible to the current API.
